x/txfees/keeper/feetokens.go

```diff
@@ -186,3 +186,22 @@ func (k Keeper) SetFeeTokens(ctx sdk.Context, feetokens []types.FeeToken) error
 	}
 	return nil
 }
+
+// SenderValidationSetFeeTokens first checks to see if the sender is whitelisted to set fee tokens.
+// If the sender is whitelisted, it sets the fee tokens.
+// If the sender is not whitelisted, it returns an error.
+func (k Keeper) SenderValidationSetFeeTokens(ctx sdk.Context, sender string, feetokens []types.FeeToken) error {
+	whitelistedAddresses := k.GetParams(ctx).WhitelistedFeeTokenSetters
+	isWhitelisted := false
+	for _, admin := range whitelistedAddresses {
+		if admin == sender {
+			isWhitelisted = true
+			break
+		}
+	}
```

Comment:
nit: could we use a helper here?

https://github.com/osmosis-labs/osmosis/blob/907fce5c98bb2aed9b289542aac7ee75918b22b9/osmoutils/slice_helper.go#L97-L105

Reply:
Use helper here https://github.com/osmosis-labs/osmosis/pull/7855/commits/9cad1cd5d90982bb5b5ab289eb9cfc08397b7df6